### PR TITLE
Sign out of all sites with Logout button on Me tab

### LINKED ISSUE #3313: Sign Out: Unable to log out of a self-hosted site
When logged directly into a self-hosted site (not via Jetpack SSO) there's no option to sign out.  The only thing I can find in the menu is to Connect to WordPress.com, but even signing in to that and then disconnecting leaves me still logged in to the .org site.

### PR DESCRIPTION
**Pull request by tonyr59h**
Fixes #3313 by replacing the existing `Disconnect from WordPress.com` button on the Me tab with a `Logout` button signs out of *all* sites, self hosted and WP.com.

A `Connect to WordPress.com` row has been added to let users who are signed into a self hosted site access their WP.com site. This new button is hidden when users are logged in. Previously the `Disconnect` button was dual purpose, connecting and disconnecting.

To test:
* Login to a self hosted site and head to the Me tab
* Verify both `Logout` and `Connect to WordPress.com` rows are visible
* Login to a WordPress.com account (without logging out of self hosted)
* Verify that only `Logout` is visible
* Select `Logout` and verify no sites are saved

cc @oguzkocer

### CODE REVIEW

File: WordPress/src/main/java/org/wordpress/android/ui/accounts/SignInFragment.java [hunk 1]
```diff
@@ -97,6 +97,7 @@
     public static final int MAX_EMAIL_LENGTH = 100;
     private static final String DOT_COM_BASE_URL = "https://wordpress.com";
     private static final String FORGOT_PASSWORD_RELATIVE_URL = "/wp-login.php?action=lostpassword";
+    private static final String XMLRPC_SUPPORT_URL = "https://codex.wordpress.org/XML-RPC_Support";
```
Comment (maxme) [inline WordPress/src/main/java/org/wordpress/android/ui/accounts/SignInFragment.java]: Looks like this is not used anymore

**Comment by maxme** [comment]
Can we drop all the changes in `SignInFragment`? It's not related to this fix.

File: WordPress/src/main/java/org/wordpress/android/ui/main/MeFragment.java [hunk 2]
```diff
@@ -265,6 +268,18 @@ public void onClick(View v) {
             }
         });
 
+        rootView.findViewById(R.id.row_logout).setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                if (mSiteStore.hasSelfHostedSite()) {
```
Comment (maxme) [inline WordPress/src/main/java/org/wordpress/android/ui/main/MeFragment.java]: We're not showing the warning dialog here.

File: WordPress/src/main/java/org/wordpress/android/ui/main/MeFragment.java [hunk 3]
```diff
@@ -254,7 +257,7 @@ public void onClick(View v) {
             }
         });
 
-        rootView.findViewById(R.id.row_logout).setOnClickListener(new View.OnClickListener() {
+        rootView.findViewById(R.id.row_wpcom_connect).setOnClickListener(new View.OnClickListener() {
             @Override
             public void onClick(View v) {
                 if (mAccountStore.hasAccessToken()) {
```
Comment (maxme) [inline WordPress/src/main/java/org/wordpress/android/ui/main/MeFragment.java]: We should change this to "If there is at least one local draft on one of their site then { show warning }" (iterate over sites / mPostStore.getPostsForSite - might be worth adding a new getter to the PostStore `hasLocalChangesOrLocalDrafts(site)` cc @aforcier )

**Comment by maxme** [comment]
We should not use the same icon for login wpcom / logout
screenshot-2017-03-23_09 09 19 196

File: WordPress/src/main/java/org/wordpress/android/ui/main/MeFragment.java (hunk 3 above)
Comment (maxme) [inline WordPress/src/main/java/org/wordpress/android/ui/main/MeFragment.java]: ^ ignore this for now, we have another ticket for this problem https://github.com/wordpress-mobile/WordPress-Android/issues/4573

**Comment by maxme** [comment]
There is a brief moment on sign out when we see both "logout" and "login" buttons: 

https://cloudup.com/cnZlmAw1xHo

It's not a big deal, but we should do the extra work to avoid this (probably something in the MeFragment.onResume call just after the progress dialog is dismissed)

File: WordPress/src/main/java/org/wordpress/android/ui/main/MeFragment.java (hunk 2 above)
Comment (tonyr59h) [inline WordPress/src/main/java/org/wordpress/android/ui/main/MeFragment.java]: af446b9e83e4c6f36744fb90ff760eef6e51af77 should do the trick.

**Comment by tonyr59h** [comment]
Let's hold off on merging this until we make sure the login icon is appropriate.

**Comment by tonyr59h** [comment]
After chatting with @mattmiklic about this flow we're thinking it needs a heavier design pass than just a login icon. Closing this PR until the UX has been OK'd.

**Pull request closed without merging**